src/legacy/ui/public/index_patterns/__mocks__/index.ts

```diff
@@ -17,12 +17,34 @@
  * under the License.
  */
 
-export function getRoutes() {
-  return {
-    edit: '/management/kibana/index_patterns/{{id}}',
-    addField: '/management/kibana/index_patterns/{{id}}/create-field',
-    indexedFields: '/management/kibana/index_patterns/{{id}}?_a=(tab:indexedFields)',
-    scriptedFields: '/management/kibana/index_patterns/{{id}}?_a=(tab:scriptedFields)',
-    sourceFilters: '/management/kibana/index_patterns/{{id}}?_a=(tab:sourceFilters)',
+import { dataPluginMock } from '../../../../core_plugins/data/public/mocks';
+
+const createMock = () => {
+  const mock = {
+    setup: {
+      data: dataPluginMock.createSetup(),
+    },
+    start: {
+      data: dataPluginMock.createStart(),
+    },
   };
-}
+  return mock;
+};
+
+createMock();
+
+// static code
+export {
+  CONTAINS_SPACES,
+  getFromSavedObject,
+  getRoutes,
+  isFilterable,
+  validateIndexPattern,
+  ILLEGAL_CHARACTERS,
+  INDEX_PATTERN_ILLEGAL_CHARACTERS,
+  INDEX_PATTERN_ILLEGAL_CHARACTERS_VISIBLE,
+  IndexPatternAlreadyExists,
+  IndexPatternMissingIndices,
+  NoDefaultIndexPattern,
+  NoDefinedIndexPatterns,
+} from '../../../../core_plugins/data/public';
```

Comment:
Since code was moved into the shimmed data plugin, a bunch of tests failed due to stuff that needed to be mocked now that the whole data plugin was getting kicked off. 

So the approach here was to mimic `jest.mock('ui/new_platform')`... basically if your plugin is still importing from the legacy location in `ui/index_patterns`, you can use the new platform mock along with `jest.mock('ui/index_patterns')`, and everything will be taken care of for you; no need to worry about what downstream dependencies the data plugin has.